everett/manager.py

```diff
@@ -71,6 +97,26 @@ def get_parser(parser):
     return parser
 
 
+def get_key_from_dict(configs, key, namespace=None):
```

Comment:
I think extracting this is a good idea. Plus it normalizes the transforms we do to keys and namespaces to get a final key.

I think using "configs" here is confusing since we typically use "config" by itself to refer to a `ConfigManager` or something similar. I think what we really want here one or more dict-like things. Maybe scopes or environments (also overloaded)? Can you think of other options?